kmip/tests/unit/services/server/test_engine.py

```diff
@@ -4361,6 +4363,248 @@ def test_activate_not_allowed_by_policy(self):
+            revocation_reason=reason_compromise,
+            compromise_date=date)
+
+        response_payload = e._process_revoke(payload)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        e._logger.info.assert_any_call(
+            "Processing operation: Revoke"
+        )
+        self.assertEqual(
+            str(object_id),
+            response_payload.unique_identifier.value
+        )
+
+        symmetric_key = e._data_session.query(
+            pie_objects.SymmetricKey
+        ).filter(
+            pie_objects.ManagedObject.unique_identifier == object_id
+        ).one()
+
+        self.assertEqual(enums.State.COMPROMISED, symmetric_key.state)
+
+        # Test that the ID placeholder can also be used to specify revocation.
+        symmetric_key.state = enums.State.ACTIVE
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+        e._id_placeholder = str(object_id)
+        payload = revoke.RevokeRequestPayload(
+            revocation_reason=reason_unspecified,
+            compromise_date=date)
+
+        response_payload = e._process_revoke(payload)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        e._logger.info.assert_any_call(
+            "Processing operation: Revoke"
+        )
+        self.assertEqual(
+            str(object_id),
+            response_payload.unique_identifier.value
+        )
+
+        symmetric_key = e._data_session.query(
+            pie_objects.SymmetricKey
+        ).filter(
+            pie_objects.ManagedObject.unique_identifier == object_id
+        ).one()
+
+        self.assertEqual(enums.State.DEACTIVATED, symmetric_key.state)
+
+    def test_revoke_on_not_active_object(self):
+        """
+        Test that the right error is generated when an revocation request is
+        received for an object that is not active with the reason other than
+        KEY_COMPROMISE.
+        """
+        e = engine.KmipEngine()
+        e._data_store = self.engine
+        e._data_store_session_factory = self.session_factory
+        e._data_session = e._data_store_session_factory()
+        e._logger = mock.MagicMock()
+
+        managed_object = pie_objects.SymmetricKey(
+            enums.CryptographicAlgorithm.AES,
+            0,
+            b''
+        )
+        e._data_session.add(managed_object)
+        e._data_session.commit()
+        e._data_session = e._data_store_session_factory()
+
+        self.assertEqual(enums.State.PRE_ACTIVE, managed_object.state)
+
+        object_id = str(managed_object.unique_identifier)
+
+        reason_unspecified = objects.RevocationReason(
+            code=enums.RevocationReasonCode.UNSPECIFIED)
+        date = primitives.DateTime(
+            tag=enums.Tags.COMPROMISE_OCCURRENCE_DATE, value=6)
+
+        # Test that reason UNSPECIFIED will put object into state
+        # DEACTIVATED
```

Comment:
So the test goal of this function is already in the function comment at top. I will just delete this comment.